SU2_CFD/src/output/CFlowOutput.cpp

```diff
@@ -143,6 +156,7 @@ void CFlowOutput::SetAnalyzeSurface(const CSolver *solver, const CGeometry *geom
   vector<su2double> Surface_VelocityIdeal     (nMarker,0.0);
   vector<su2double> Surface_Area              (nMarker,0.0);
   vector<su2double> Surface_MassFlow_Abs      (nMarker,0.0);
+  vector<su2double> Surface_Species           (nMarker*nSpecies,0.0); // (Y0_bound0, Y0_bound1,... , Y1_bound0, Y1_bound1,...)
```

Comment:
use a matrix type please

Reply:
I went with it this because of the MPI communication below  (which is done using other containers which follow the same logic). I did not have to worry whether that works there ... so first make it work and then make it nice :) 

Is `vector<vector<su2double> >` ok? And I guess I have to flatten it before the MPI-comms and fluff if up afterwards. my shallow midnight searches did not really enlighten me there

Reply:
no, su2activematrix

Reply:
Done in 194b213

I added another lambda for the Allreduce part. Unfortunately [lambdas cant be templated in C++11](https://stackoverflow.com/questions/3575901/can-lambda-functions-be-templated)